storm/src/main/storm/mesos/MesosNimbus.java

```diff
@@ -577,288 +409,267 @@ public void assignSlots(Topologies topologies, Map<String, Collection<WorkerSlot
       for (WorkerSlot slot : topologyToSlots.getValue()) {
         TopologyDetails details = topologies.getById(topologyId);
         LOG.debug("assignSlots: topologyId: {} worker being assigned to slot: {} with workerCpu: {} workerMem: {}",
-                  topologyId, slot, MesosCommon.topologyWorkerCpu(_conf, details), MesosCommon.topologyWorkerMem(_conf, details));
+                  topologyId, slot, MesosCommon.topologyWorkerCpu(mesosStormConf, details), MesosCommon.topologyWorkerMem(mesosStormConf, details));
       }
     }
 
     synchronized (_offersLock) {
-      computeLaunchList(topologies, slotsForTopologiesNeedingAssignments);
+      /**
+       * We need to call getConsolidatedOfferResourcesPerNode again here for the following
+       *   1. Because _offers could have changed between `allSlotsAvailableForScheduling()' and `assignSlots()'
+       *   2. In `allSlotsAvailableForScheduling()', we change what is returned by `getConsolidatedOfferResourcesPerNode'
```

Comment:
```
 /**
       *  We need to call getConsolidatedOfferResourcesPerNode again here for the following reasons
       *    1. Because _offers could have changed between `allSlotsAvailableForScheduling()' and `assignSlots()'
       *    2. In `allSlotsAvailableForScheduling()', we change what is returned by `getConsolidatedOfferResourcesPerNode' in order to calculate the number of slots available.
       */
```